dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -62,13 +64,63 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
         }
     }
 
+    void PushNotificationManager::RegisterUnpackagedApplicationHelper(
+        const winrt::guid& remoteId,
+        _Out_ wil::unique_cotaskmem_string &unpackagedAppUserModelId)
+    {
+        wchar_t processName[1024];
+        THROW_HR_IF(ERROR_FILE_NOT_FOUND, GetModuleFileNameExW(GetCurrentProcess(), NULL, processName, sizeof(processName) / sizeof(processName[0])) == 0);
+
+        THROW_IF_FAILED(::CoInitializeEx(nullptr, COINITBASE_MULTITHREADED));
+
+        auto scopeExit = wil::scope_exit(
+            [&]() { CoUninitialize(); });
+
+        wil::com_ptr<INotificationsLongRunningPlatform> notificationPlatform{
+            wil::CoCreateInstance<NotificationsLongRunningPlatform, INotificationsLongRunningPlatform>(CLSCTX_LOCAL_SERVER) };
+
+        winrt::check_hresult(notificationPlatform->RegisterFullTrustApplication(processName, remoteId, &unpackagedAppUserModelId));
+    }
+
+    winrt::hresult PushNotificationManager::CreateChannelWithRemoteIdHelper(const winrt::guid& remoteId, ChannelDetails& channelInfo)
```

Comment:
Should this method be returning a `std::optional<ChannelDetails>` instead of using an out-reference-param and returning an hresult?

Reply:
For Now I would like to go by the returning hresult here as based on that - whether it is S_OK, E_NOTIMPL or other errors I prefer the caller to make decisions.

Reply:
I have opened a bug for this -> #1365